### PR TITLE
fix:[PLG-373] Update Tenable data collector and policies

### PR DESCRIPTION
**Pull request by mkinchin-paladin**
# Description

Now collecting:
* Only assets with last_authenticated_scan_time is not null
* Vulnerabilities with plugin type is local
Now policy checks if a VM was scanned by authenticated scan any time in the past
If Tenable knows about that instance and scanned it (VMsScannedByTenableRule didn't fail) - we check for found vulns.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Did run collection and policies locally:
Result:
Screenshot 2023-12-05 at 6 43 20 PM
"Enabled scan" policy failed for not scanned VMs
"Vulns found" failed for those with found vulns

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] My commit message/PR follows the contribution guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

### CODE REVIEW

File: jobs/pacman-tenable-enricher/src/main/java/com/tmobile/cso/pacman/tenable/Main.java [hunk 1]
```diff
@@ -66,13 +66,13 @@ public static void execute(Map<String, String> params) {
...
-            if (!errorInfo.isEmpty() && errorInfo.get("error") != null) {
-                log.warn("Job executed with some errors -> {}", errorInfo);
+            if (!errorInfo.isEmpty() && errorInfo.get("errors") != null && ((List<Map<String, Object>>) errorInfo.get("errors")).size() > 0) {
+                log.error("Job executed with some errors -> {}", errorInfo);
             } else {
                 log.info("Job executed successfully");
             }
         } else {
-            log.warn("Job hint is not supplied");
+            log.error("Job hint is not supplied");
```
Comment (plyubich) [inline jobs/pacman-tenable-enricher/src/main/java/com/tmobile/cso/pacman/tenable/Main.java:75]: Why is an error? It means that there is not data returned but it's a valid response from open search?

File: jobs/pacman-tenable-enricher/src/main/java/com/tmobile/cso/pacman/tenable/jobs/TenableDataImporter.java [hunk 2]
```diff
@@ -59,16 +59,16 @@ protected TenableDataImporter() {
     public abstract Map<String, Object> execute(int days);
 
     private void getTenableInfo() {
-        String secretManagerPrefix = System.getProperty("secret.manager.path"); // paladincloud/secret
-        String baseAccount = System.getProperty("base.account");
-        String baseRegion = System.getProperty("base.region");
-        String roleName = System.getProperty("s3.role");
+//        String secretManagerPrefix = System.getProperty("secret.manager.path"); // paladincloud/secret
```
Comment (plyubich) [inline jobs/pacman-tenable-enricher/src/main/java/com/tmobile/cso/pacman/tenable/jobs/TenableDataImporter.java:62]: Why keep this?

**Review by plyubich — COMMENTED** [review]
LGTM

File: jobs/pacman-tenable-enricher/src/main/java/com/tmobile/cso/pacman/tenable/Main.java (hunk 1 above)
Comment (mkinchin-paladin) [reply jobs/pacman-tenable-enricher/src/main/java/com/tmobile/cso/pacman/tenable/Main.java:75]: This goes not from OS, it hoes from job definition, as a param. So if it's not provided then job was not configured properly during installation (see screenshot)
image

**Pull request merged**